arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -1178,12 +1178,68 @@ IResearchAnalyzerFeature::AnalyzerPool::ptr IResearchAnalyzerFeature::get( // fi
               << "failure creating an arangosearch static analyzer instance "
                  "for name '"
               << name << "'";
-          throw irs::illegal_state();  // this should never happen, treat as an
-                                       // assertion failure
+
+          // this should never happen, treat as an assertion failure
+          THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL);
```

Comment:
For "internal" exceptions, we should always add a descriptive error message. Otherwise there will be just an exception with "internal error. please report this error to arangodb.com" or so.
```suggestion
          THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "failed to create arangosearch static analyzer");
```